protocol-units/bridge/shared/tests/abstract_blockchain.rs

```diff
@@ -0,0 +1,163 @@
+use bridge_shared::types::{
+	Amount, BridgeTransferDetails, BridgeTransferId, GenUniqueHash, HashLock, InitiatorAddress,
+	RecipientAddress, TimeLock,
+};
+use bridge_shared::types::{HashLockPreImage, LockDetails};
+use futures::StreamExt;
+use rand::{Rng, SeedableRng};
+use rand_chacha::ChaChaRng;
+
+use test_log::test;
+
+mod shared;
+
+use shared::testing::blockchain::{
+	AbstractBlockchain, AbstractBlockchainEvent, CounterpartyCall, InitiatorCall, Transaction,
+};
+
+use crate::shared::testing::blockchain::{
+	counterparty_contract::SmartContractCounterpartyEvent,
+	initiator_contract::SmartContractInitiatorEvent,
+};
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+struct TestAddress(pub &'static str);
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+struct TestHash(pub &'static str);
+
+impl From<TestAddress> for RecipientAddress {
+	fn from(value: TestAddress) -> Self {
+		RecipientAddress(value.0.as_bytes().to_vec())
+	}
+}
+
+impl From<RecipientAddress> for TestAddress {
+	fn from(value: RecipientAddress) -> Self {
+		Self(static_str_ops::staticize(&String::from_utf8(value.0).expect("Invalid UTF-8")))
+	}
+}
+
+impl From<HashLockPreImage> for TestHash {
+	fn from(_value: HashLockPreImage) -> Self {
+		todo!()
+	}
+}
+
+impl GenUniqueHash for TestHash {
+	fn gen_unique_hash<R: Rng>(_rng: &mut R) -> Self {
+		TestHash("unique_hash")
+	}
+}
+
+#[test(tokio::test)]
+async fn test_initiate_bridge_transfer() {
+	let rng = ChaChaRng::from_seed([0u8; 32]);
+	let mut blockchain = AbstractBlockchain::<TestAddress, TestHash, _>::new(rng, "TestBlockchain");
+
+	let mut monitor = blockchain.add_event_listener();
+
+	let initiator_address = InitiatorAddress(TestAddress("initiator"));
+	let recipient_address = RecipientAddress::from(TestAddress("recipient"));
+	let amount = Amount(1000);
+	let time_lock = TimeLock(100);
+	let hash_lock = HashLock(TestHash("hash_lock"));
+
+	let transaction = Transaction::Initiator(InitiatorCall::InitiateBridgeTransfer(
+		initiator_address.clone(),
+		recipient_address.clone(),
+		amount,
+		time_lock.clone(),
+		hash_lock.clone(),
+	));
+
+	blockchain.transaction_sender.unbounded_send(transaction).unwrap();
```

Comment:
I've commented this on other folks PRs. But, I prefer that we avoid unwraps even in tests. In tests because people often use tests as examples and copy code accordingly.

Reply:
I understand your perspective. The code in question is designed primarily for testing purposes, and it is not intended for broader use. The potential failure mode identified would predominantly occur due to insufficient memory. Converting this to an `expect` could be considered, although the necessity of such a change might be minimal. However, if you deem the modification essential, I am prepared to implement this adjustment. Please advise on how you wish to proceed.

Reply:
I will comment it again on a later PR or put it in an issue. I suppose we're going ahead with the merge here to unblock @0xmovses